torch/distributed/_shard/replicated_tensor.py

```diff
@@ -73,11 +73,15 @@ def dispatch_arg(arg):
                 all_replicated = False
 
         for arg in args:
-            dispatch_arg(arg)
+            redispatched_res = dispatch_arg(arg)
+            if redispatched_res:
+                return redispatched_res
```

Comment:
I'm not sure if this would work as expected always. For example, for things like in-place ops there is no return value. In such cases, don't we still erroneously go to kwargs?

Maybe `dispatch_arg` should return a tuple of return value and whether the function was executed or not.

Reply:
addressed.